### PR TITLE
Issue 3332

### PR DESCRIPTION
**Pull request by farzad-sedghi**
FIX https://github.com/spotify/scio/issues/3332

This way (advanced) users can provide a lambdas (`configOverride` below) to modify the underlying PTransforms, where they are using BigQueryIO syntax methods. Same approach can possibly be used for all other IOs.
This map function is applied to the ptransform **as the last step before applying the ptransform to the data**.

```
def main(cmdlineArgs: Array[String]): Unit = {
    ...
    implicit def fun: Write[MyTypes.Output] => Write[MyTypes.Output] =
        w => w.withTableDescription("Table description set by overriding PTransform in the job")
     ...
    sc
      .parallelize(List((1,11), (2, 22)))
      .map(kv => MyTypes.Output(kv._1.intValue, kv._2.intValue()))
      .saveAsTypedBigQueryTable(
        table = Table.Spec("scio-playground:farzaddev.test6"),
        writeDisposition = WRITE_TRUNCATE,
        createDisposition = CREATE_IF_NEEDED,
        configOverride = _.withTableDescription("Table description set by overriding PTransform in the job")
      )
    sc.run()
    ()
  }

```

### CODE REVIEW

File: scio-google-cloud-platform/src/main/scala/com/spotify/scio/bigquery/syntax/SCollectionSyntax.scala [hunk 1]
```diff
@@ -51,14 +51,17 @@ final class SCollectionTableRowOps[T <: TableRow](private val self: SCollection[
     createDisposition: CreateDisposition = BigQueryTypedTable.WriteParam.DefaultCreateDisposition,
     tableDescription: String = BigQueryTypedTable.WriteParam.DefaultTableDescription,
     timePartitioning: TimePartitioning = BigQueryTypedTable.WriteParam.DefaultTimePartitioning
+  )(
+    implicit writePTransformOverride: Write[TableRow] => Write[TableRow] = identity
```
Comment (RustedBones) [inline scio-google-cloud-platform/src/main/scala/com/spotify/scio/bigquery/syntax/SCollectionSyntax.scala:55]: I don't think this should be an implicit parameter

File: scio-google-cloud-platform/src/main/scala/com/spotify/scio/bigquery/BigQueryIO.scala [hunk 2]
```diff
@@ -215,41 +217,45 @@ object BigQueryTypedTable {
...
-  trait WriteParam {
+  trait WriteParam[T] {
     val schema: TableSchema
     val writeDisposition: WriteDisposition
     val createDisposition: CreateDisposition
     val tableDescription: String
     val timePartitioning: TimePartitioning
     val extendedErrorInfo: ExtendedErrorInfo
     val insertErrorTransform: SCollection[extendedErrorInfo.Info] => Unit
+    val writePTransformOverride: Write[T] => Write[T]
```
Comment (RustedBones) [inline scio-google-cloud-platform/src/main/scala/com/spotify/scio/bigquery/BigQueryIO.scala:228]: I's suggest `configure` as name

**Comment by shnapz** [comment]
@farzad-sedghi we need to merge it to [v0.13.x](https://github.com/spotify/scio/tree/v0.13.x)

**Comment by shnapz** [comment]
@farzad-sedghi ooops, now we need to merge latest master changes to 0.13.x

**Comment by RustedBones** [comment]
@farzad-sedghi I think this can be closed in favor of #4810

**Pull request closed without merging**